src/Bookings.js

```diff
@@ -1,18 +1,77 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import SearchResults from "./SearchResults.js";
+import CustomerProfile from "./CustomerProfile.js";
 
 const Bookings = () => {
+  const [bookings, setBookings] = useState([]);
+  const [customerId, setCustomerId] = useState(null);
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [errorOccurred, setErrorOccurred] = useState({
+    ok: true,
+    status: "",
+    statusText: ""
+  });
+
+  useEffect(() => {
+    fetch(`https://cyf-react.glitch.me/delayed`)
+      .then(res => {
+        if (!res.ok) {
+          setErrorOccurred({
+            ok: res.ok,
+            status: res.status,
+            statusText: res.statusText
+          });
+          throw new Error(res);
+        } else {
+          return res.json();
+        }
+      })
+      .then(data => {
+        setBookings(data);
+        setData(data);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        console.log(error);
+      });
+  }, []);
+
   const search = searchVal => {
-    console.info("TO DO!", searchVal);
+    if (searchVal.trim() === "") {
+      setBookings(data);
+    } else {
+      const filterBookings = bookings.filter(element => {
+        return (
+          element.firstName.toLowerCase().startsWith(searchVal.toLowerCase()) ||
+          element.surname.toLowerCase().startsWith(searchVal.toLowerCase())
+        );
+      });
+      setBookings(filterBookings);
+    }
   };
 
   return (
     <div className="App-content">
       <div className="container">
         <Search search={search} />
-        {/* <SearchResults results={FakeBookings} /> */}
+        {errorOccurred.ok === false ? (
+          <div>
+            <span>
+              An Error occurred when fetching the Bookings Data :{" "}
+              {errorOccurred.status} {errorOccurred.statusText}
+            </span>
+          </div>
+        ) : isLoading === true ? (
```

Comment:
Do we need to check if this `isLoading` boolean is equal to true here?  It only has two possible values

Reply:
Same with `ok` above 😄